src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.File.Tests.cs

```diff
@@ -71,6 +71,49 @@ public void SetsLastModifiedTimeOnExtractedFiles()
             Assert.InRange(File.GetLastWriteTime(outFile).Ticks, dt.AddSeconds(-3).Ticks, dt.AddSeconds(3).Ticks); // include some slop for filesystem granularity
         }
 
+        [Fact]
+        public void SetsLastModifiedTimeOnExtractedDirectories()
+        {
+            using TempDirectory root = new TempDirectory();
+
+            DirectoryInfo fromDir = Directory.CreateDirectory(Path.Combine(root.Path, "fromdir"));
+            // Create a hierarcy of directories.
+            var directories = new DirectoryInfo[]
+            {
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir")),                      // 'fromdir/dir'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir", "child")),             // 'fromdir/dir/child'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir", "child", "subchild")), // 'fromdir/dir/child/subchild'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir2")),                     // 'fromdir/dir2'
+                Directory.CreateDirectory(Path.Combine(fromDir.FullName, "dir2", "child2")),           // 'fromdir/dir2/child'
+            };
+            var dt = new DateTime[directories.Length];
+            for (int i = directories.Length - 1; i >= 0; i--) // Reverse order to preserve parent timestamps.
+            {
+                // Add a file.
+                File.Create(Path.Combine(directories[i].FullName, "file")).Dispose();
+
+                // Set the directory timestamp.
+                dt[i] = new DateTime(2000 + i, 1, 2, 3, 4, 5, DateTimeKind.Local);
+                directories[i].LastWriteTime = dt[i];
```

Comment:
@danmoseley the directory last write times get set here.

Reply:
sorry I wasn't clear -- I meant in the extraction code.

Reply:
It is calling `TarEntry.AttemptSetLastWriteTime`.

That calls `File.SetLastWriteTime`. Maybe Windows doesn't like this for directories ... :crossed_fingers: